.github/workflows/ci.yaml

```diff
@@ -404,30 +403,29 @@ jobs:
         brew uninstall --ignore-dependencies libffi 2>&1 || true
 
     - name: build/test wheels
-      id: build
+      uses: pypa/cibuildwheel@v2.18.1
       env:
         CIBW_BUILD: ${{ matrix.spec }}
         CIBW_PRERELEASE_PYTHONS: 'True'
         CIBW_TEST_REQUIRES: pytest setuptools
         CIBW_TEST_COMMAND: pip install pip --upgrade; cd {project}; PYTHONUNBUFFERED=1 pytest
         MACOSX_DEPLOYMENT_TARGET: ${{ matrix.deployment_target || '10.9' }}
         SDKROOT: ${{ matrix.sdkroot || 'macosx' }}
-      run: |
-        set -eux
-        
-        mkdir cffi
-        
-        tar zxf ${{ steps.fetch_sdist.outputs.download-path }}/cffi*.tar.gz/cffi*.tar.gz --strip-components=1 -C cffi
+      with:
+        package-dir: >-
```

Comment:
*(`cibuildwheel` unarchives sdist tarballs if supplied in place of a directory)*

Reply:
I'm likely going to move the libffi build to its own cached-artifact job though, which means we'll still have to give cibuildwheel a full directory to get the cached artifact dir into the container envs alongside the sdist- that's why I abandoned just giving cibuildwheel the sdist tarball directly. I looked over the code- it didn't look like there was a way to build directly from an sdist tarball *and* send a project directory with build deps at the same time.

Reply:
Any reason those things shouldn't just be in the sdist? It's supposed to be self-contained.